modules/adot/variables.tf

```diff
@@ -39,8 +39,10 @@ variable "adot_config" {
   description = "accept_namespace_regex defines the list of namespaces from which metrics will be exported, and additional_metrics defines additional metrics to export."
   type        = any
   default = {
+    default_config         = true
     accept_namespace_regex = "(default|kube-system)"
     additional_metrics     = []
+    template               = null
```

Comment:
maybe better to have name "helm_values" or "values_yaml"  instead having it named just "template"? as this is actually full content of adot helm chart values.yaml, which can be generated from outside based on template file or even passed directly as just a yaml string,

also seems no need for "default_config" bool field as you can identify if default used by checking if values.yaml content is passed or not

please have description for this field, you can use a single comment for this new field or have in primary description